jetty-xml/src/test/java/org/eclipse/jetty/xml/XmlConfigurationTest.java

```diff
@@ -1855,6 +1859,43 @@ public void testVirtualCallWithClassAttribute() throws Exception
         assertThat(tc.testObject, instanceOf(ScheduledExecutorService.class));
     }
 
+    public static Stream<Arguments> xmlSystemIdSource()
+    {
+        List<Arguments> ids = new ArrayList<>();
+
+        String[] schemes = {"http", "https"};
+        String[] hosts = {"eclipse.org", "www.eclipse.org", "eclipse.dev", "www.eclipse.dev"};
+        String[] paths = {"/jetty/configure.dtd", "/jetty/configure_9_3.dtd", "/jetty/configure_10_0.dtd"};
+
+        for (String scheme: schemes)
+        {
+            for (String host: hosts)
+            {
+                for (String path: paths)
+                {
+                    ids.add(Arguments.of(String.format("%s://%s%s", scheme, host, path)));
+                }
+            }
+        }
+
+        return ids.stream();
+    }
+
+    /**
+     * Test to ensure that all the XML System ID variants are covered in the
+     * {@link XmlConfiguration} internals.
+     */
+    @ParameterizedTest
+    @MethodSource("xmlSystemIdSource")
+    public void testSystemIdVariants(String xmlSystemId) throws IOException, SAXException
+    {
+        // empty raw xml, just to instantiate XmlConfiguration, so we can access the ConfigurationParser.
+        XmlConfiguration xmlConfiguration = asXmlConfiguration("<Configure class=\"org.eclipse.jetty.xml.TestConfiguration\" />");
+        XmlConfiguration.ConfigurationParser configurationProcessor = xmlConfiguration.getParser();
```

Comment:
I would rename `getParser()` to `getXmlParser()` and return type to be `XmlParser`, which is already a public class.

Reply:
Done